### ISSUE TITLE
WebSocket is already in CLOSING or CLOSED state.

### ISSUE THREAD

**Issue by olzii** [pion/ion-sfu#638]
### Your environment.
  - Version: 8e1eaa5
  - Client: Vue.JS and "ion-sdk-js": "^1.7.2",
  - Environement:  Ubuntu 20.04, Go version : go version go1.17.6 linux/amd64
  - Are you using a TURN server?  Yes:  stun:stun.l.google.com:19302

### What did you do?
I am trying this example ([Pion/Ion-Sfu One to Many](https://github.com/jason-shen/ion-one-to-many-vuejs)) and deployed ION SFU and Vue example on real server.

### What happened?

SFU and WebSocket is working without any problem about 1 minute and closing automatically. 
And saying this error on browser console : 'WebSocket is already in CLOSING or CLOSED state.'
![Screenshot 2022-01-13 213920](https://user-images.githubusercontent.com/5012289/149362828-bc0fa2fc-bd6a-4b3a-ae08-8ce8cd438823.JPG)

On server side, it has following log:
![Screenshot 2022-01-13 214011](https://user-images.githubusercontent.com/5012289/149363545-c7bf6679-666b-406a-a1c0-4f8ab39237b4.JPG)
![Screenshot 2022-01-13 214059](https://user-images.githubusercontent.com/5012289/149363579-9e473564-4f6a-4737-ac2c-406fe6c6d77f.JPG)

**Comment by Y1nPengfei** [comment]
me too.
![image](https://user-images.githubusercontent.com/30924624/149438424-e785df30-382c-4370-9bbe-8e9df1a0a7b4.png)

**Comment by olzii** [comment]
> @olzii are you using a reverse proxy for the websockets like nginx?, if so please make sure that the timeout is long enough or you need to handle gracefully the ws reconnec.

Yes, i am using Nginx proxy. Default proxy timeout was 60 seconds, lets try to change it.

![image](https://user-images.githubusercontent.com/5012289/149439624-870bf4f5-c823-4810-b821-5bed214dae3f.png)

**Comment by olzii** [comment]
I have changed timeout to 310 seconds: 

proxy_read_timeout 310s;

After 5 minute 10 seconds, it received Websocket closed error again.

**Comment by Y1nPengfei** [comment]
> I have changed timeout to 310 seconds:
> 
> proxy_read_timeout 310s;
> 
> After 5 minute 10 seconds, it received Websocket closed error again.

nat1to1 is public ip？

**Comment by olzii** [comment]
Yes it is public ip.

Here is my Nginx proxy config : 

`location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $http_host;
        proxy_set_header X-NginX-Proxy true;
        proxy_pass http://192.168.10.99:7000;
        proxy_redirect off;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_read_timeout 310s;
    }
`

**Comment by Y1nPengfei** [comment]
> Yes it is public ip.
> 
> Here is my Nginx proxy config :
> 
> ```
> location / {
>         proxy_set_header X-Real-IP $remote_addr;
>         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
>         proxy_set_header Host $http_host;
>         proxy_set_header X-NginX-Proxy true;
>         proxy_pass http://192.168.10.99:7000;
>         proxy_redirect off;
>         proxy_http_version 1.1;
>         proxy_set_header Upgrade $http_upgrade;
>         proxy_set_header Connection "upgrade";
>         proxy_read_timeout 310s;
>     }
> ```
I shut it down because of the ICE problem.  The solution is to change the value of configuration file NAT1to1 to public ip.Other reasons for the closure have not yet been encountered

**Comment by olzii** [comment]
@Y1nPengfei 

My client and server using same ICE server 

ION-SFU:
![image](https://user-images.githubusercontent.com/5012289/149445182-e8e0f150-d7d8-43aa-a1f9-6e1b0c85d704.png)
Vue:
![image](https://user-images.githubusercontent.com/5012289/149445235-23494d9e-2dd7-43b8-9151-6af50e4e832c.png)

Also i am using NAT1to1 config with my public IP.
![image](https://user-images.githubusercontent.com/5012289/149445561-b303af2e-1397-472c-b150-73973be5a657.png)

**Comment by Y1nPengfei** [comment]
> @Y1n鹏飞
> 
> 我的客户端和服务器使用相同的 ICE 服务器
> 
> ION-SFU： Vue： ![图片](https://user-images.githubusercontent.com/5012289/149445182-e8e0f150-d7d8-43aa-a1f9-6e1b0c85d704.png) ![图片](https://user-images.githubusercontent.com/5012289/149445235-23494d9e-2dd7-43b8-9151-6af50e4e832c.png)
> 
> 我也在使用带有我的公共 IP 的 NAT1to1 配置。 ![图片](https://user-images.githubusercontent.com/5012289/149445561-b303af2e-1397-472c-b150-73973be5a657.png)


![image](https://user-images.githubusercontent.com/30924624/149445909-e32996a0-4859-4c81-b52a-b40e8939b291.png)
Try commenting out ICelite or setting it to false

**Comment by olzii** [comment]
@Y1nPengfei 

Making "icelite=false" has no effect, result is same.

**Comment by olzii** [comment]
[NGINX to reverse proxy websockets AND enable SSL](https://stackoverflow.com/questions/12102110/nginx-to-reverse-proxy-websockets-and-enable-ssl-wss)

If i increase proxy_read_timeout  value to 86400, it is working. 
But i think it is not solving the issue, it just making timeout longer

**Comment by ahmed-adly-khalil** [comment]
I have the exact same issue, anyone found a solution? i'm using NGINX and Kubernetes + the connection is over https

**Comment by OrlandoCo** [comment]
The solution is simple, just increase the timeout or gracefully recover from websocket reconnect.
